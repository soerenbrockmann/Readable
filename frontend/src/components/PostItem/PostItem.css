.postitem {
  display: flex;
  flex-direction: column;
  border: 1px solid #f8f8f8;
  padding: 20px;
  background-color: #f8f8f8;
  border-bottom: 1px solid #c8c8c8;
}

.title {
  font-weight: bold;
  margin-bottom: 5px;
}

.body {
  margin-bottom: 10px;
}

.postitemnav {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  margin-bottom: 10px;
  flex-wrap: wrap;
  margin-top: 1em;
}

.postitemnav > div {
  margin-right: 0.5em;
}

.postitemnav > div:nth-child(3) {
  margin-left: auto;
}

.votescore {
  margin-right: 1em;
  color: #3399ff;
}

.commentcount {
  color: #3399ff;
}

.postitemheader {
  display: flex;
  flex-direction: row;
}

.postitemheader > div {
  margin-right: 0.5em;
}

.postitemheader > div:nth-child(2) {
  margin-left: auto;
}
