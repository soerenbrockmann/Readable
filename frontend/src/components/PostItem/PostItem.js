import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import Votes from "../Votes";
import "./PostItem.css";
import FormatDate from "../FormatDate";

const PostItem = ({
  onDelete,
  onEdit,
  onDownVote,
  onUpVote,
  post: { timestamp, title, body, author, category, voteScore, commentCount },
  to,
  isClosedBtn = false,
  isCommentCount = false
}) => {
  return (
    <div className="postitem">
      <div className="postitemheader">
        {to && (
          <Link to={to}>
            <div className="title">{title}</div>
          </Link>
        )}
        {!to && <div className="title">{title}</div>}
        <div>
          <Link to={"/"} className="ondelete" onClick={onDelete}>
            Delete
          </Link>
        </div>
        <div className="onedit" onClick={onEdit}>
          Edit
        </div>
        {isClosedBtn && (
          <div>
            <Link to={"/"} className="onclose">
              Close
            </Link>
          </div>
        )}
      </div>
      <div className="meta">
        <div>{author} - </div> <FormatDate timestamp={timestamp} />
      </div>
      <div className="body">
        {body}
        {to && (
          <Link to={to}>
            <div className="readmore">...Read more</div>
          </Link>
        )}
      </div>
      <div className="postitemnav">
        <Votes
          onUpVote={onUpVote}
          onDownVote={onDownVote}
          voteScore={voteScore}
        />
        {isCommentCount && (
          <Link to={to}>
            <div className="commentcount">{commentCount} Comments</div>
          </Link>
        )}
        <div className="category">[{category}]</div>
      </div>
    </div>
  );
};

FormatDate.propTypes = {
  timestamp: PropTypes.number.isRequired
};

Votes.propTypes = {
  onUpVote: PropTypes.func.isRequired,
  onDownVote: PropTypes.func.isRequired,
  voteScore: PropTypes.number
};

export default PostItem;
