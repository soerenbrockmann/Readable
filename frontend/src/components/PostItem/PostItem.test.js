import React from "react";
import { shallow } from "enzyme";
import PostItem from "./PostItem";

describe("<PostItem />", () => {
  const item = {
    timestamp: 1468166872634,
    title: "",
    body: "",
    author: "",
    category: "",
    voteScore: 1,
    commentCount: 2
  };

  const postItem = shallow(
    <PostItem
      post={item}
      onDelete={jest.fn()}
      onEdit={jest.fn()}
      onDownVote={jest.fn()}
      onUpVote={jest.fn()}
      to={"/"}
    />
  );
  it("renders properly", () => {
    expect(postItem).toMatchSnapshot();
  });
});
