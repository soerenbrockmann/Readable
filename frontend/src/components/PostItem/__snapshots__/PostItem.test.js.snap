// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItem /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostItem
    onDelete={[Function]}
    onDownVote={[Function]}
    onEdit={[Function]}
    onUpVote={[Function]}
    post={
        Object {
            "author": "",
            "body": "",
            "category": "",
            "commentCount": 2,
            "timestamp": 1468166872634,
            "title": "",
            "voteScore": 1,
          }
    }
    to="/"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="postitemheader"
>
          <Link
                    replace={false}
                    to="/"
          >
                    <div
                              className="title"
                    />
          </Link>
          <div>
                    <Link
                              className="ondelete"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                    >
                              Delete
                    </Link>
          </div>
          <div
                    className="onedit"
                    onClick={[Function]}
          >
                    Edit
          </div>
</div>,
        <div
          className="meta"
>
          <div>
                    
                     - 
          </div>
           
          <FormatDate
                    timestamp={1468166872634}
          />
</div>,
        <div
          className="body"
>
          
          <Link
                    replace={false}
                    to="/"
          >
                    <div
                              className="readmore"
                    >
                              ...Read more
                    </div>
          </Link>
</div>,
        <div
          className="postitemnav"
>
          <Votes
                    onDownVote={[Function]}
                    onUpVote={[Function]}
                    voteScore={1}
          />
          <div
                    className="category"
          >
                    [
                    
                    ]
          </div>
</div>,
      ],
      "className": "postitem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
>
              <div
                            className="title"
              />
</Link>,
            false,
            <div>
              <Link
                            className="ondelete"
                            onClick={[Function]}
                            replace={false}
                            to="/"
              >
                            Delete
              </Link>
</div>,
            <div
              className="onedit"
              onClick={[Function]}
>
              Edit
</div>,
            false,
          ],
          "className": "postitemheader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="title"
/>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "title",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="ondelete"
                onClick={[Function]}
                replace={false}
                to="/"
>
                Delete
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete",
                "className": "ondelete",
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Delete",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit",
              "className": "onedit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit",
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              
               - 
</div>,
            " ",
            <FormatDate
              timestamp={1468166872634}
/>,
          ],
          "className": "meta",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                " - ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "",
              " - ",
            ],
            "type": "div",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "timestamp": 1468166872634,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <Link
              replace={false}
              to="/"
>
              <div
                            className="readmore"
              >
                            ...Read more
              </div>
</Link>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="readmore"
>
                ...Read more
</div>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "...Read more",
                "className": "readmore",
              },
              "ref": null,
              "rendered": "...Read more",
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Votes
              onDownVote={[Function]}
              onUpVote={[Function]}
              voteScore={1}
/>,
            false,
            <div
              className="category"
>
              [
              
              ]
</div>,
          ],
          "className": "postitemnav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onDownVote": [Function],
              "onUpVote": [Function],
              "voteScore": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "[",
                "",
                "]",
              ],
              "className": "category",
            },
            "ref": null,
            "rendered": Array [
              "[",
              "",
              "]",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="postitemheader"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <div
                                    className="title"
                        />
            </Link>
            <div>
                        <Link
                                    className="ondelete"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/"
                        >
                                    Delete
                        </Link>
            </div>
            <div
                        className="onedit"
                        onClick={[Function]}
            >
                        Edit
            </div>
</div>,
          <div
            className="meta"
>
            <div>
                        
                         - 
            </div>
             
            <FormatDate
                        timestamp={1468166872634}
            />
</div>,
          <div
            className="body"
>
            
            <Link
                        replace={false}
                        to="/"
            >
                        <div
                                    className="readmore"
                        >
                                    ...Read more
                        </div>
            </Link>
</div>,
          <div
            className="postitemnav"
>
            <Votes
                        onDownVote={[Function]}
                        onUpVote={[Function]}
                        voteScore={1}
            />
            <div
                        className="category"
            >
                        [
                        
                        ]
            </div>
</div>,
        ],
        "className": "postitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
>
                <div
                                className="title"
                />
</Link>,
              false,
              <div>
                <Link
                                className="ondelete"
                                onClick={[Function]}
                                replace={false}
                                to="/"
                >
                                Delete
                </Link>
</div>,
              <div
                className="onedit"
                onClick={[Function]}
>
                Edit
</div>,
              false,
            ],
            "className": "postitemheader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="title"
/>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "title",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="ondelete"
                  onClick={[Function]}
                  replace={false}
                  to="/"
>
                  Delete
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Delete",
                  "className": "ondelete",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Delete",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "onedit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                
                 - 
</div>,
              " ",
              <FormatDate
                timestamp={1468166872634}
/>,
            ],
            "className": "meta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  " - ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                " - ",
              ],
              "type": "div",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "timestamp": 1468166872634,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <Link
                replace={false}
                to="/"
>
                <div
                                className="readmore"
                >
                                ...Read more
                </div>
</Link>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="readmore"
>
                  ...Read more
</div>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "...Read more",
                  "className": "readmore",
                },
                "ref": null,
                "rendered": "...Read more",
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Votes
                onDownVote={[Function]}
                onUpVote={[Function]}
                voteScore={1}
/>,
              false,
              <div
                className="category"
>
                [
                
                ]
</div>,
            ],
            "className": "postitemnav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onDownVote": [Function],
                "onUpVote": [Function],
                "voteScore": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "[",
                  "",
                  "]",
                ],
                "className": "category",
              },
              "ref": null,
              "rendered": Array [
                "[",
                "",
                "]",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
